src/Microsoft.Framework.Runtime.Abstractions/Library.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Framework.Runtime
+{
+    /// <summary>
+    /// Exposes information about a library which can be an assembly, project, or a package.
+    /// </summary>
+    public sealed class Library
+    {
+        /// <summary>
+        /// Gets the name of the library.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        ///  Gets the version of the library.
+        /// </summary>
+        public string Version { get; }
+
+        /// <summary>
+        /// Gets the path to the library. For projects, this is a path to the project.json file.
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// Gets the type of library. Common values include Project, Package, and Assembly.
+        /// </summary>
+        public string Type { get; }
+
+        /// <summary>
+        /// Gets a list of dependencies for the library. The dependencies are names are other <see cref="ILibrary"/> objects.
+        /// </summary>
+        public IEnumerable<string> Dependencies { get; }
+
+        /// <summary>
+        /// Gets a list of assembly names from the library that can be loaded. Packages can contain multiple assemblies.
+        /// </summary>
+        public IEnumerable<AssemblyName> LoadableAssemblies { get; }
```

Comment:
Since we're renaming this whole type, lets change this name as well to Assemblies. We might also want to expose the paths to the assemblies. Lets discuss this as part of this change.